sphinx/tutorial/rose/furthertopics/rose-stem.rst

```diff
@@ -0,0 +1,309 @@
+and ``fire_lasers``.
+
+The scheduling section contains the Jinja2 code to use the information we
+have already set to generate the graph based on what the user
+requested on the command line.
+
+The runtime section should be familiar. Note, however, that the
+``fcm_make_spaceship`` task sets the environment variable
+``SOURCE_SPACESHIP`` from the Jinja2 variable of the same name. This
+is how the variables passed with ``--source`` on the command line are
+passed to ``fcm-make``, which then uses these environment variables in
+its own configuration files.
+
+
+The ``rose-suite.conf`` file
+----------------------------
+
+The suites associated with ``rose-stem`` require a version number
+indicating the version of the ``rose stem`` command with which they
+are compatible. This is specified in the ``rose-suite.conf`` file,
+together with the default values of ``RUN_NAMES`` and ``SOURCE_SPACESHIP``.
+Paste the following into your ``rose-suite.conf`` file:
+
+.. code-block:: rose
+
+   ROSE_STEM_VERSION=1
+
+   [jinja2:suite.rc]
+   RUN_NAMES=[]
+   SOURCE_SPACESHIP='fcm:spaceship_tr@head'
+
+Both of the Jinja2 variables will be overriden by the user when they
+execute ``rose stem`` on the command line.
+
+
+The ``rose_ana_position`` app
+-----------------------------
+
+The final component is a ``rose ana`` app to test the position of our
+spaceship matches the correct output.
+
+Create an app named ``rose_ana_position`` and paste the following into its
+``rose-app.conf`` file:
+
+.. code-block:: rose
+
+   [ana:grepper.FilePattern(Check X position at each timestep)]
+   pattern='^\s*Position:\s*(.*?)\s*,'
+   files=/home/user/spaceship/kgo.txt
+        =../spaceship/output.txt
+
+   [ana:grepper.FilePattern(Check Y position at each timestep)]
+   pattern='^\s*Position:.*?,\s*(.*?)\s*,'
+   files=/home/user/spaceship/kgo.txt
+        =../spaceship/output.txt
+
+   [ana:grepper.FilePattern(Check Z position at each timestep)]
+   pattern='^\s*Position:.*,\s*(.*)\s*$'
+   files=/home/user/spaceship/kgo.txt
+        =../spaceship/output.txt
+
+This will check that the positions reported by the program match those
+within the known good output file.
+
+
+Known Good Output
+-----------------
+
+In the root of the working copy is a file called ``kgo.txt``.
+
+Replace the ``/home/user/spaceship`` paths in the ``rose_ana_position``
+app with the path to this file.
+
+The known good output should be the result of a control run. ``rose ana``
+will compare the answers from this file (obtained using the extract and
+comparison methods in the ``rose-app.conf`` file) with the results from
+the user's code change.
+
+
+Adding the suite to version control
+-----------------------------------
+
+Before running the suite we need to make sure that all the files and
+directories we have created are known to the version control system.
+
+Add all the new files you've created using ``fcm add -c`` *(answer yes
```

Comment:
Maybe explain briefly that the files are added automatically through that exact command, i.e. we don't have to go back through the tutorial, pick out all new files and do ``fcm add -c <new-file>`` as I originally thought might be the case.